registration/src/main/java/sample/lobby/registration/projection/ConferenceProjection.java

```diff
@@ -0,0 +1,210 @@
+        incrementState(conference.build());
+    }
+
+    private boolean seatTypesListContains(SeatTypeId id) {
+        final List<SeatType> filtered = filterById(id, getState().getSeatTypeList());
+        final boolean contains = !filtered.isEmpty();
+        return contains;
+    }
+
+    @Subscribe
+    public void on(SeatTypeUpdated event) {
+        final Conference.Builder conference = getState().toBuilder();
+        final SeatType newSeatType = event.getSeatType();
+        final SeatTypeId id = newSeatType.getId();
+        final List<SeatType> seatTypes = conference.getSeatTypeList();
+
+        final List<SeatType> filtered = filterById(id, seatTypes);
+        if (filtered.isEmpty()) {
+            log().warn("No seat type with ID %s; conference ID: %s", id.getUuid(), conference.getId().getUuid());
+            return;
+        }
+        final SeatType oldSeatType = filtered.get(0);
+        final int oldTypeIndex = seatTypes.indexOf(oldSeatType);
+        conference.setSeatType(oldTypeIndex, newSeatType);
+
+        addOrRemoveSeats(newSeatType.getQuantityTotal(), oldSeatType.getQuantityTotal(), id);
+
+        incrementState(conference.build());
+    }
+
+    private void addOrRemoveSeats(int newQuantity, int oldQuantity, SeatTypeId id) {
+        final int difference = newQuantity - oldQuantity;
+        if (difference > 0) {
+            sendAddSeatsRequest(id, difference);
+        } else if(difference < 0) {
+            sendRemoveSeatsRequest(id, abs(difference));
+        }
+    }
+
+    private void sendAddSeatsRequest(SeatTypeId seatTypeId, int quantity) {
+        final AddSeats command = AddSeats.newBuilder()
+                .setConferenceId(getState().getId())
+                .setSeatTypeId(seatTypeId)
+                .setQuantity(quantity)
+                .build();
+        final CommandRequest request = newCommandRequest(command, CommandContext.getDefaultInstance());
+        boundedContext.process(request);
+    }
+
+    private void sendRemoveSeatsRequest(SeatTypeId seatTypeId, int quantity) {
+        final RemoveSeats command = RemoveSeats.newBuilder()
+                .setConferenceId(getState().getId())
+                .setSeatTypeId(seatTypeId)
+                .setQuantity(quantity)
+                .build();
+        final CommandRequest request = newCommandRequest(command, CommandContext.getDefaultInstance());
+        boundedContext.process(request);
+    }
+
+    private static List<SeatType> filterById(final SeatTypeId id, List<SeatType> seatTypes) {
+        final Iterable<SeatType> result = filter(seatTypes, new Predicate<SeatType>() {
+            @Override
+            public boolean apply(@Nullable SeatType input) {
+                return (input != null) && input.getId().equals(id);
+            }
+        });
+        return ImmutableList.copyOf(result);
+    }
+
+    protected static Conference convert(sample.lobby.contracts.conference.Conference c) {
```

Comment:
It is the conference from another module. I'll extract it from both these modules to one.